### ISSUE TITLE
[Main thread] Dev status

### ISSUE THREAD

**Issue by qwreey75** [qwreey75/quick-settings-tweaks#81]
# Bugs that i can fix now

## ~~Fix fade effect~~ (Fixed)
<del>
#19

Event `notify::vscrollbar-visible` can follow scrollbar visible status

![image](https://user-images.githubusercontent.com/46598063/234870523-d73adaa9-dc98-4793-a17c-58c366858a4e.png)

```js
r(57).connect("notify::vscrollbar-visible",()=>log("QLOG vscrollbar-visible"))
```
https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/fc587b3e0c08b09e0121bc2dd319cef6cdb4c340/src/st/st-scroll-view.c#L706

and we can update overlay_scrollbars to fix fade effect issue

![image](https://user-images.githubusercontent.com/46598063/234871103-ec531e99-4d57-4fb1-be9c-bbe24b21c90e.png)

</del>

## ~~Quick toggle attached to the bottom of qs~~ (Fixed)

<del>
If some extensions add a quick toggle after the qsTweaks extension is loaded, that quick toggle will be attached to bottom of the qs Panel
![image](https://user-images.githubusercontent.com/46598063/234783277-070c9193-e13c-4f0c-a19b-da50948980de.png)

We can use actor-removed actor-added on st-widget to check children added / removed
![image](https://user-images.githubusercontent.com/46598063/234875777-c2679296-00a0-4fbc-93a3-eefa0bf0219e.png)

</del>

## ~~Reappearing hidden buttons~~ (Fixed)

<del>
#73 #65

notify::mapped can check menu is opened, closed
r(0).connect("notify::mapped",()=>log("QLOG asdf"))
![image](https://user-images.githubusercontent.com/46598063/234880050-47786ce1-cae3-4551-a058-b9a2319c2bdf.png)

We can recheck children and update visible status when menu opened

</del>

# Other

## I can implement

1. #67
2. #45 ?Maybe?
3. #62 ?Maybe?
4. #4 ?Maybe?

## Bug that i can't reproduce, fix

1. #63
5. #68

## Bug that I found

1. weird position of notification time
![image](https://user-images.githubusercontent.com/46598063/234887675-0a23fddb-d819-461b-bd92-b95c841bdbd9.png)

**Comment by qwreey75** [comment]
Now use GObject.insert_child_at_index which have more performence benefit instead of Me.imports.libs.utility.addChildWithIndex

**Comment by qwreey75** [comment]
Feature idea:

gnome unsafe mode quick toggle (for gnome dev)

**Comment by qwreey75** [comment]
Now, using `busctl --user call org.gnome.Shell /org/gnome/Shell org.gnome.Shell Eval s 'Meta.restart("Restarting…", global.context)'` to reload gnome shell in `make dev`. this is would better for speed, and stability

**Comment by qwreey75** [comment]
Now, Extension loader no longer uses GLib.timeout_add. Instead, just init features immediately.

Because timeout make errors invisible, debug harder

**Comment by qwreey75** [comment]
datemenu-fix-weather-widget no longer used, i'll remove

**Comment by qwreey75** [comment]
Fixed https://github.com/qwreey75/quick-settings-tweaks/issues/19

**Comment by Ridhubharan** [comment]
The quick toggle for "Caffeine" gnome extension appearing below the media player is fixed by the developer.

**Comment by qwreey75** [comment]
> The developer fixed the quick toggle for "Caffeine" gnome extension appearing below the media player.
in https://github.com/eonpatapon/gnome-shell-extension-caffeine/pull/250 PR, it was fixed

But, anyway, QSTweaks doesn't update it self index, that means if any extension, not only Caffeine add QSToggle, it may appear below the notification box

**Comment by qwreey75** [comment]
> Feature idea:
> 
> gnome unsafe mode quick toggle (for gnome dev)

![image](https://user-images.githubusercontent.com/46598063/235737963-76be348e-c1d7-4ff5-9c7e-07a5ac8f32cd.png)

**Comment by qwreey75** [comment]
Maybe i can fix #63 a bit by adding a scroll
